test/e2e/upgrade_management_cluster_test.go

```diff
@@ -2,340 +2,137 @@ package e2e
-		// If we do not go that route, we should instead refactor the whole
-		// upgradeManagementCluster function into smaller parts.
-		upgradeIronic(upgradeClusterProxy.GetClientSet())
-		upgradeBMO(upgradeClusterProxy.GetClientSet())
-
-		/*-------------------------------*
-		| Upgrade the management cluster |
-		*--------------------------------*/
-		By("Upgrading providers to the latest version available")
-		By("Get the management cluster images before upgrading")
-		printImages(upgradeClusterProxy)
-
-		Logf("Upgrade management cluster to : %v", clusterv1.GroupVersion.Version)
-		clusterctl.UpgradeManagementClusterAndWait(ctx, clusterctl.UpgradeManagementClusterAndWaitInput{
-			ClusterctlConfigPath: clusterctlConfigPath,
-			ClusterProxy:         upgradeClusterProxy,
-			Contract:             clusterv1.GroupVersion.Version,
-			LogFolder:            filepath.Join(artifactFolder, "clusters", workloadClusterName),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		Logf("The target cluster upgraded!")
-
-		By("Get the management cluster images after upgrading")
-		printImages(upgradeClusterProxy)
-
-		By("Upgrade bootstrap cluster")
-		By("Get the management cluster images before upgrading")
-		printImages(bootstrapClusterProxy)
-
-		Logf("Upgrade bootstrap cluster to : %v", clusterv1.GroupVersion.Version)
-		clusterctl.UpgradeManagementClusterAndWait(ctx, clusterctl.UpgradeManagementClusterAndWaitInput{
-			ClusterctlConfigPath: clusterctlConfigPath,
-			ClusterProxy:         bootstrapClusterProxy,
-			Contract:             clusterv1.GroupVersion.Version,
-			LogFolder:            filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		Logf("The source cluster upgraded!")
-
-		By("Get the bootstrap cluster images after upgrading")
-		printImages(bootstrapClusterProxy)
-
-		By("UPGRADE MANAGEMENT CLUSTER PASSED!")
-	})
-
-	AfterEach(func() {
-		dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, e2eConfig.GetIntervals, clusterName, clusterctlLogFolder, skipCleanup)
-	})
+		os.Setenv("BAREMETALV4_POOL_RANGE_START", "192.168.111.201")
+		os.Setenv("BAREMETALV4_POOL_RANGE_END", "192.168.111.240")
+		os.Setenv("PROVISIONING_POOL_RANGE_START", "172.22.0.201")
+		os.Setenv("PROVISIONING_POOL_RANGE_END", "172.22.0.240")
+	}
 }
 
-func printImages(clusterProxy framework.ClusterProxy) {
-	pods, err := clusterProxy.GetClientSet().CoreV1().Pods("").List(ctx, metav1.ListOptions{})
-	Expect(err).To(BeNil())
-	var images []string
-	for _, pod := range pods.Items {
-		for _, c := range pod.Spec.Containers {
-			exist := false
-			for _, i := range images {
-				if i == c.Image {
-					exist = true
-					break
-				}
-			}
-			if !exist {
-				images = append(images, c.Image)
-				Logf("%v", c.Image)
-			}
-		}
+func getPreUpgrade(ctx context.Context) func(proxy framework.ClusterProxy) {
```

Comment:
Same. This also needs a better name and doc string.

Reply:
I still want to keep same generic names for call hooks added docs